binary_search_tree/tree.py

```diff
@@ -2,58 +2,116 @@ class TreeNode:
     def __init__(self, key, val = None):
         if val == None:
             val = key
-
         self.key = key
         self.value = val
         self.left = None
         self.right = None
         
-
-
 class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
-    def add(self, key, value = None):
-        pass
+    def add_helper(self, current_root, new_node):
+      if current_root is None:
+          return new_node
+      if new_node.key < current_root.key:
+          current_root.left = self.add_helper(current_root.left, new_node)
+      else:
+          current_root.right = self.add_helper(current_root.right, new_node)
+      return current_root
+      
+    def find_helper (self, key, current_node):
+        if current_node == None:
+            return None
+        if current_node.key == key:
+            return current_node.value
+        if key < current_node.key:
+            return self.find_helper(key, current_node.left)
+        if key > current_node.key:
+            return self.find_helper(key, current_node.right)
+    
+    def inorder_helper(self, current, items):
```

Comment:
Minor thing: `inorder_helper`, `preorder_helper`, and `postorder_helper` are all very similar methods, but `inorder_helper` has the `current` parameter before the `items` parameter, while the other two have it the other way.

This can trip up someone trying to call the methods so it may make sense  to match the order across all three. (As well as matching the names so `current` is `current_node` like the other two.)

Reply:
The implementations of all of them are fine, though!